invenio_base/app.py

```diff
@@ -178,6 +188,25 @@ def cli(**params):
     return cli
 
 
+def finalize_app_loader(app, entry_points=None):
+    """Run functions before of the first request.
+
+    This loader is the last possible position where it is possible to configure the app.
+
+    NOTE: it replaces the before_first_request decorator of flask <2.3.0
+
+    :param entry_points: List of entry points providing to Flask extensions.
+
+    .. versionadded: 2.0.0
+    """
+
+    def loader_init_func(func):
+        with app.app_context():
+            func(app)
```

Comment:
```suggestion
        func(app)
```

major: it might actually be possible to remove this. The only really blocking module we found in terms of relying on application context is `invenio-pages` (because it performs an SQL query to fetch the available pages from the database).

Reply:
Opened https://github.com/inveniosoftware/invenio-base/issues/177